src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs

```diff
@@ -184,10 +184,11 @@ public void UpdateDescriptionForModel(MappingDataFormatParameter mappingSource)
             //The dimension is the first from the supported dimension which
             //has the selected unit.
             column.Unit = _mappingParameterEditorPresenter.Unit;
-            column.Dimension = _columnInfos
-               .First(x => x.DisplayName == model.MappingName)
-               .SupportedDimensions
-               .FirstOrDefault(x => x.HasUnit(column.Unit.SelectedUnit));
+            if (column.Dimension != null && !column.Dimension.HasUnit(column.Unit.SelectedUnit))
+               column.Dimension = _columnInfos
+                  .First(x => x.DisplayName == model.MappingName)
+                  .SupportedDimensions
+                  .FirstOrDefault(x => x.HasUnit(column.Unit.SelectedUnit));
```

Comment:
I see this code over an over . FirstOrDefault(x=>x.HasUnit). It needs to be refactored at some point. I am pointing out here because I am seeing it. Please create an issue

Reply:
it's always coming from the SupportedDimensions. If we make it a real class as opposed to a list? Or it belongs to the Columninfo itself. Not that Supporteddimension is a List and not a IReadOnlyList which breaks encapsulation

Reply:
#1520